data/transactions/logic/debugger_legacy.go

```diff
@@ -0,0 +1,376 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package logic
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+// LegacyDebuggerHook is an interface that supports the first version of TEAL debuggers.
+// It consists of a set of functions called by eval function during TEAL program execution
+type LegacyDebuggerHook interface {
+	// Register is fired on program creation
+	Register(state *DebugState) error
+	// Update is fired on every step
+	Update(state *DebugState) error
+	// Complete is called when the program exits
+	Complete(state *DebugState) error
+}
+
+type legacyDebuggerAdaptor struct {
+	debugger      LegacyDebuggerHook
+	innerTxnDepth int
+	debugState    *DebugState
+}
+
+// MakeLegacyDebuggerAdaptor creates an adaptor that externally adheres to the DebuggerHook
+// interface, but drives a LegacyDebuggerHook debugger
+func MakeLegacyDebuggerAdaptor(debugger LegacyDebuggerHook) DebuggerHook {
+	return &legacyDebuggerAdaptor{debugger: debugger}
+}
+
+// BeforeTxn does nothing
+func (adaptor *legacyDebuggerAdaptor) BeforeTxn(ep *EvalParams, groupIndex int) error {
+	return nil
+}
+
+// AfterTxn does nothing
+func (adaptor *legacyDebuggerAdaptor) AfterTxn(ep *EvalParams, groupIndex int) error {
+	return nil
+}
```

Comment:
By embedding NullDebuggerHook these become unnecessary.
(I'm also hoping `NullDebuggerHook` becomes `NullEvalTracer`, but that's not my point here.)